app/uk/gov/hmrc/helptosavefrontend/controllers/EmailController.scala

```diff
@@ -107,49 +110,57 @@ class EmailController @Inject() (val helpToSaveService:          HelpToSaveServi
 
     }(redirectOnLoginURL = routes.EmailController.getSelectEmailPage().url)
 
-  def selectEmailSubmit(): Action[AnyContent] =
-    authorisedForHtsWithInfo { implicit request ⇒ implicit htsContext ⇒
-
-      def handleForm(email: String, eligibilityCheckResult: Option[Either[Ineligible, EligibleWithUserInfo]], bankDetails: Option[BankDetails], backLink: Option[String]): Future[Result] =
-          SelectEmailForm.selectEmailForm.bindFromRequest().fold(
-            withErrors ⇒ Ok(views.html.email.select_email(email, withErrors, backLink)),
-            _.newEmail.fold[Future[Result]](
-
-              SeeOther(routes.EmailController.confirmEmail(crypto.encrypt(email)).url))(
-                newEmail ⇒ {
-                  updateSessionAndReturnResult(HTSSession(eligibilityCheckResult, None, Some(newEmail), None, None, bankDetails),
-                                               SeeOther(routes.EmailController.verifyEmail().url)
-                  )
-                }
-              )
-          )
+  def selectEmailSubmit(): Action[AnyContent] = authorisedForHtsWithInfo { implicit request ⇒ implicit htsContext ⇒
+
+    def handleForm(email:    String,
+                   backLink: Option[String],
+                   session:  HTSSession
+      ): Future[Result] =
+        SelectEmailForm.selectEmailForm.bindFromRequest().fold(
+          withErrors ⇒ Ok(views.html.email.select_email(email, withErrors, backLink)),
+          _.newEmail.fold[Future[Result]]{
+            SeeOther(routes.EmailController.confirmEmail(crypto.encrypt(email)).url)
+          } { newEmail ⇒
+            val updatedSession = session.copy(pendingEmail   = Some(newEmail), confirmedEmail = None)
+
+            if (updatedSession =!= session) {
+              updateSessionAndReturnResult(session.copy(pendingEmail   = Some(newEmail), confirmedEmail = None),
```

Comment:
`session.copy(pendingEmail   = Some(newEmail), confirmedEmail = None)` is this not same as `updatedSession` declared above ?

Reply:
whoops. I missed that

Reply:
fixed and pushed now